rosetta/configuration/errors.go

```diff
@@ -19,16 +19,26 @@ import (
 )
 
 var (
-	ErrorInternal                 = meta.ErrorDefinition{Code: 1, Message: "internal error", Retriable: false}
-	ErrorInvalidEncoding          = meta.ErrorDefinition{Code: 2, Message: "invalid request encoding", Retriable: false}
-	ErrorInvalidFormat            = meta.ErrorDefinition{Code: 3, Message: "invalid request format", Retriable: false}
-	ErrorInvalidNetwork           = meta.ErrorDefinition{Code: 4, Message: "invalid network identifier", Retriable: false}
-	ErrorInvalidAccount           = meta.ErrorDefinition{Code: 5, Message: "invalid account identifier", Retriable: false}
-	ErrorInvalidCurrency          = meta.ErrorDefinition{Code: 6, Message: "invalid currency identifier", Retriable: false}
-	ErrorInvalidBlock             = meta.ErrorDefinition{Code: 7, Message: "invalid block identifier", Retriable: false}
-	ErrorInvalidTransaction       = meta.ErrorDefinition{Code: 8, Message: "invalid transaction identifier", Retriable: false}
-	ErrorUnknownBlock             = meta.ErrorDefinition{Code: 9, Message: "unknown block identifier", Retriable: true}
-	ErrorUnknownCurrency          = meta.ErrorDefinition{Code: 10, Message: "unknown currency identifier", Retriable: false}
-	ErrorUnknownTransaction       = meta.ErrorDefinition{Code: 11, Message: "unknown block transaction", Retriable: false}
+	ErrorInternal           = meta.ErrorDefinition{Code: 1, Message: "internal error", Retriable: false}
+	ErrorInvalidEncoding    = meta.ErrorDefinition{Code: 2, Message: "invalid request encoding", Retriable: false}
+	ErrorInvalidFormat      = meta.ErrorDefinition{Code: 3, Message: "invalid request format", Retriable: false}
+	ErrorInvalidNetwork     = meta.ErrorDefinition{Code: 4, Message: "invalid network identifier", Retriable: false}
+	ErrorInvalidAccount     = meta.ErrorDefinition{Code: 5, Message: "invalid account identifier", Retriable: false}
+	ErrorInvalidCurrency    = meta.ErrorDefinition{Code: 6, Message: "invalid currency identifier", Retriable: false}
+	ErrorInvalidBlock       = meta.ErrorDefinition{Code: 7, Message: "invalid block identifier", Retriable: false}
+	ErrorInvalidTransaction = meta.ErrorDefinition{Code: 8, Message: "invalid transaction identifier", Retriable: false}
+	ErrorUnknownBlock       = meta.ErrorDefinition{Code: 9, Message: "unknown block identifier", Retriable: true}
+	ErrorUnknownCurrency    = meta.ErrorDefinition{Code: 10, Message: "unknown currency identifier", Retriable: false}
+	ErrorUnknownTransaction = meta.ErrorDefinition{Code: 11, Message: "unknown block transaction", Retriable: false}
+
+	// Construction API specific errors.
 	ErrorInvalidTransactionIntent = meta.ErrorDefinition{Code: 12, Message: "invalid transaction intent", Retriable: false}
+	ErrorInvalidAuthorizers       = meta.ErrorDefinition{Code: 13, Message: "invalid transaction authorizers", Retriable: false}
```

Comment:
Not sure if I was overzealous here.

I feel we're leaking knowledge here - Rosetta has no knowledge of the anatomy of a Flow transaction - the authorizers, the Cadence script, payers, proposers etc..

But the errors are precise and really tell us what went wrong.

Reply:
No, I think that is good. The Rosetta API might have no idea about the semantic meaning of a transaction's internals, but the consumer _has_ to know. Otherwise, how would he sign a transaction, right? So I think this is definitely valuable :ok_hand: